### PR TITLE
feat(products): add pagination on products page

### LINKED ISSUE #221: Show created product
When a product is created, we cannot see the product created.
chrome_hGPE6Y7urO
In this example I added a product with name Ronnan, but not appear in the list

### PR DESCRIPTION
**Pull request by eduardoquispe**
### Description

Please include a summary of the change.

fixes # (issue)

#### Screenshots

If applicable, add screenshots about your changes

#### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How to test it

If applicable, describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

### Checklist:

- [ ] Doesn't break the current code.
- [ ] Passes linters and test, also is building.
- [ ] Doesn't have spelling or grammatical problems.
- [ ] Doesn't have unnecessary comments or debugging code.
- [ ] The branch is updated with the last dev branch changes.

### CODE REVIEW

File: views/Products/List/index.tsx [hunk 1]
```diff
@@ -3,14 +3,21 @@ import { FC } from 'react'
...
+      <div className={styles['products-list']}>
+        {products.map((product: Product) => (
+          <ProductItem key={`product-${product.id}`} product={product} />
+        ))}
+      </div>
     </div>
-  </div>
-)
+  )
+}
```
Comment (Israel-Laguan) [inline views/Products/List/index.tsx]: I think this is a regression

@Eghinner please check

File: pages/productos/index.tsx [hunk 2]
```diff
@@ -1,54 +1,49 @@
...
+  products_aggregate: {
+    aggregate: {
+      totalCount: number
+    }
+  }
+}
 
 const Products: NextPage = () => {
   const [products, setProducts] = useState<Product[]>([])
+  const [totalProducts, setTotalProducts] = useState(0)
```
Comment (Israel-Laguan) [inline pages/productos/index.tsx:20]: I don't think this is really needed, use products.length instead

Comment (eduardoquispe) [reply pages/productos/index.tsx:20]: this is to save the total number of products stored in the DB, in the response there are only 15 products per page.

Comment (Israel-Laguan) [reply pages/productos/index.tsx:20]: AH got it, you are right

File: views/Products/List/index.tsx (hunk 1 above)
Comment (Israel-Laguan) [reply views/Products/List/index.tsx]: @eduardo-quispe what about leaving as it was?

Comment (eduardoquispe) [reply views/Products/List/index.tsx]: @Israel-Laguan  updated

**Review by Israel-Laguan — APPROVED** [review]

**Comment by angelik0828** [comment]
## ⚠️Please check how the pagination is, it is overlapping the products grid

image

**Comment by eduardoquispe** [comment]
I will check it now

**Comment by eduardoquispe** [comment]
already updated

File: styles/Products.module.scss [hunk 3]
```diff
@@ -1,9 +1,10 @@
 @import 'colors';
 
 .container {
-  margin: 1.5rem auto;
+  flex-basis: 85%;
+  overflow-y: auto;
   padding: 0 15px;
-
+
```
Comment (Israel-Laguan) [inline styles/Products.module.scss]:
```suggestion

```

**Review by Israel-Laguan — APPROVED** [review]

File: styles/Products.module.scss (hunk 3 above)
Comment (eduardoquispe) [reply styles/Products.module.scss]: corrected

**Review by Israel-Laguan — APPROVED** [review]

File: pages/productos/index.tsx [hunk 4]
```diff
@@ -1,54 +1,49 @@
...
-      setIsLoading(false)
-    }
 
-    getProducts()
+  const getProducts = useCallback(async (limit: number, offset: number) => {
+    const providerId = sessionStorage.getItem('providerId')
+    const { data } = await axios.get<ProductResponse>('/api/providers/products', {
+      headers: {
+        providerId: providerId || '',
+      },
```
Comment (Israel-Laguan) [inline pages/productos/index.tsx]: Does this really work? I suggest using a query param instead

@e-azocar

Comment (eduardoquispe) [reply pages/productos/index.tsx]: resolved

**Comment by Israel-Laguan** [comment]
@angelik0828 please check

**Review by Eghinner — APPROVED** [review]
Great

**Comment by angelik0828** [comment]
## ✔️ Pagination styling has been improved great job

image

**Review by angelik0828 — APPROVED** [review]

**Pull request merged**